src/content/devices/input-output/js/test.js

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+// This is a basic test file for use with testling.
+// The test script language comes from tape.
+/* jshint node: true */
+var test = require('tape');
+
+var webdriver = require('selenium-webdriver');
+var seleniumHelpers = require('../../../../../test/selenium-lib');
+
+test('Fake device selection and check video tag dimensions in ' +
+    'input-output demo', function(t) {
+  // FIXME: use env[SELENIUM_BROWSER] instead?
+  var driver = seleniumHelpers.buildDriver();
+
+  var webrtcDetectedBrowser = null;
+
+  driver.get('file://' + process.cwd() +
+      '/src/content/devices/input-output/index.html')
+  .then(function() {
+    t.pass('Page loaded');
+    return driver.executeScript('return webrtcDetectedBrowser;');
+  })
+  .then(function(browser) {
+    webrtcDetectedBrowser = browser;
+    // Making sure we can select the 1st audio device.
+    // TODO: Select more devices if Firefox adds a 2nd fake A/V device and
+    // Chrome adds another fake video device.
+    t.pass('Selecting 1st audio device');
+    var option = driver.findElement(webdriver.By.xpath(
+        'id("audioSource")/option[1]'));
+    option.click();
+    return option.getAttribute('value');
+  })
+  .then(function(deviceId) {
+    // Check enumerateDevices has returned an id.
+    t.ok(deviceId !== '', 'Device/source id: ' + deviceId);
+  })
+  .then(function() {
+    // Making sure we can select the 1st video device.
+    // TODO: Select more devices if Firefox adds a 2nd fake A/V device and
+    // Chrome adds another fake video device.
+    t.pass('Selecting 1st video device');
+    var option = driver.findElement(webdriver.By.xpath(
+        'id("videoSource")/option[1]'));
+    option.click();
+    return option.getAttribute('value');
```

Comment:
how about chaining this? i.e. 
```return driver.findElement(...)
.click()
.getAttribute('value')
```
Also above. I have a slight preference for that style since it doesn't appear to be synchronous while it actually isn't. See https://code.google.com/p/selenium/wiki/WebDriverJs#Value_Propagation_and_Chaining